### PR TITLE
Feature/api auth

### PR DESCRIPTION
**Pull request by sondreb**
- Adds support for API key authentication (disabled by default) for REST API.
- Add protection to all APIs by default, a few APIs is made public.

### CODE REVIEW

File: src/Blockcore/Blockcore.csproj [hunk 1]
```diff
@@ -27,6 +27,7 @@
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
+    <PackageReference Include="Swashbuckle.AspNetCore.Annotations" Version="5.5.1" />
```
Comment (dangershony) [inline src/Blockcore/Blockcore.csproj:30]: IS this required in the core lib or can it go to the nodes  or dedicated feature?

File: src/Blockcore/Controllers/ConnectionManagerController.cs [hunk 2]
```diff
@@ -7,14 +7,17 @@
...
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.Extensions.Logging;
+using Swashbuckle.AspNetCore.Annotations;
 
 namespace Blockcore.Controllers
 {
     /// <summary>
     /// A <see cref="FeatureController"/> that implements API and RPC methods for the connection manager.
     /// </summary>
+    [Authorize]
```
Comment (dangershony) [inline src/Blockcore/Controllers/ConnectionManagerController.cs:20]: Seems correct, will this also be required for localhost?

File: src/Features/Blockcore.Features.NodeHost/Authentication/AppSettingsGetApiKeyQuery.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using Blockcore.Features.NodeHost.Settings;
+using Microsoft.Extensions.Options;
+
+namespace Blockcore.Features.NodeHost.Authentication
+{
+    public class AppSettingsGetApiKeyQuery : IGetApiKeyQuery
+    {
+        private BlockcoreSettings settings;
```
Comment (dangershony) [inline src/Features/Blockcore.Features.NodeHost/Authentication/AppSettingsGetApiKeyQuery.cs:10]: Shouldn't this be the nodehostsettings?

File: src/Features/Blockcore.Features.NodeHost/NodeHostSettings.cs [hunk 4]
```diff
@@ -40,6 +40,11 @@ public class NodeHostSettings
         /// </summary>
         public bool EnableAPI { get; private set; }
 
+        /// <summary>
+        /// If true will require authentication on all sensitive APIs. Some APIs will be public available.
+        /// </summary>
+        public bool EnableAuth { get; private set; }
```
Comment (dangershony) [inline src/Features/Blockcore.Features.NodeHost/NodeHostSettings.cs:46]: ah pretty cool ok so this is off by default

Comment (dangershony) [reply src/Features/Blockcore.Features.NodeHost/NodeHostSettings.cs:46]: I wonder if we should try to distinguish localhost and  as well

File: src/Blockcore/Blockcore.csproj (hunk 1 above)
Comment (sondreb) [reply src/Blockcore/Blockcore.csproj:30]: It is needed to add tags, but the tags was not shown in the UI anyway, so I'll remove both the tags and this package.

File: src/Blockcore/Controllers/ConnectionManagerController.cs (hunk 2 above)
Comment (sondreb) [reply src/Blockcore/Controllers/ConnectionManagerController.cs:20]: There won't be any difference where traffic comes from.

File: src/Features/Blockcore.Features.NodeHost/Authentication/AppSettingsGetApiKeyQuery.cs (hunk 3 above)
Comment (sondreb) [reply src/Features/Blockcore.Features.NodeHost/Authentication/AppSettingsGetApiKeyQuery.cs:10]: nodehostsettings already exists and is a ini-file based configuration. This is a complex type that is serialized from the appsettings.json file.

File: src/Features/Blockcore.Features.NodeHost/NodeHostSettings.cs (hunk 4 above)
Comment (sondreb) [reply src/Features/Blockcore.Features.NodeHost/NodeHostSettings.cs:46]: No reason to distinguish localhost.

**Pull request merged**